draft-ietf-ppm-dap.md

```diff
@@ -1200,34 +1190,100 @@ enum {
+splitting each report into report shares, one for each Aggregator. The Leader
+and Helper then run the aggregate initialization flow to accomplish two tasks:
+
+1. Recover and determine which input report shares are valid.
+1. For each valid report share, initialize the VDAF preparation process (see
+   {{Section 5.2 of !VDAF}}).
+
 The Leader and Helper initialization behavior is detailed below.
 
 #### Leader Initialization {#leader-init}
 
 The Leader begins the aggregate initialization phase with the set of candidate
-report shares as follows:
+reports as follows:
 
-1. Generate a fresh AggregationJobID. The ID value MUST be unique within the
-   scope of the corresponding DAP task.
+1. Generate a fresh AggregationJobID.
 1. Decrypt the input share for each report share as described in
    {{input-share-decryption}}.
 1. Check that the resulting input share is valid as described in
    {{input-share-validation}}.
-1. Initialize VDAF preparation as described in {{input-share-prep}}.
 
-If any step invalidates the report share, the Leader removes the report share
-from the set of candidate reports. Once the leader has initialized this state
-for all valid candidate report shares, it creates an `AggregationJobInitReq`
-message for each Helper to initialize the preparation of this candidate set. The
-`AggregationJobInitReq` message is structured as follows:
+If any step invalidates the report, the Leader rejects the report and removes
+it from the set of candidate reports.
+
+Next, for each report the Leader executes the following procedure:
+
+~~~
+state = VDAF.ping_pong_start(vdaf_verify_key,
+                             True,
+                             agg_param,
+                             report_id,
+                             public_share,
+                             plaintext_input_share.payload)
+if state != Rejected():
+  (state, outbound) = VDAF.ping_pong_req(agg_param, state, None)
+~~~
+
+where:
+
+* `vdaf_verify_key` is the VDAF verification key for the task
+* `agg_param` is the VDAF aggregation parameter provided by the Collector (see
+  {{collect-flow}})
+* `report_id` is the report ID, used as a nonce
+* `public_share` is the report's public share
+* `plaintext_input_share` is the Leader's `PlaintextInputShare`
+
+The methods are defined in {{!VDAF}}. This process determines the initial
+per-report `state`, as well as the initial `outbound` message to send to the
+Helper. (These are coalesced into a single HTTP request to the Helper as
+described below.) If `state == Rejected()`, then the report is rejected and
+removed from the set of candidate reports.
+
+[[OPEN ISSUE #470: update references to the ping-pong API, once a VDAF draft is
+published which describes it.]]
+
+Next, for each candidate report the Leader constructs a `PrepareInit` message
+structured as follows:
 
 ~~~
 struct {
   ReportMetadata report_metadata;
   opaque public_share<0..2^32-1>;
   HpkeCiphertext encrypted_input_share;
```

Comment:
I think we should continue to define the `ReportShare` struct, even if we don't need to. This way it's clear what we mean by "report share".
```suggestion
  ReportShare report_share;
```

Reply:
What if we just rename `struct PrepareInit` to `struct ReportShare`?